packages/headless/src/features/generated-answer/generated-answer-analytics-actions.ts

```diff
@@ -0,0 +1,70 @@
+import {
+  AnalyticsType,
+  CustomAction,
+  SearchAction,
+  makeAnalyticsAction,
+} from '../analytics/analytics-utils';
+import {
+  citationSourceSelector,
+  generativeQuestionAnsweringIdSelector,
+} from './generated-answer-selectors';
+
+export const logRetryGeneratedAnswer = (): SearchAction =>
+  makeAnalyticsAction(
+    'analytics/generatedAnswer/retry',
+    AnalyticsType.Search,
+    (client) => client.makeRetryGeneratedAnswer()
+  );
+
+export const logOpenGeneratedAnswerSource = (
+  citationId: string
+): CustomAction =>
+  makeAnalyticsAction(
+    'analytics/generatedAnswer/openSource',
```

Comment:
nit: openSource -> clickSource to avoid confusion :) ?

Reply:
I hadn't even thought that just 'openSource' could lead to confusion. 😅